opm/autodiff/FlowMain.hpp

```diff
@@ -391,6 +391,8 @@ namespace Opm
                 deck_ = parser->parseFile(deck_filename, parseContext);
                 checkDeck(deck_, parser);
                 eclipse_state_.reset(new EclipseState(deck_, parseContext));
+                auto ioConfig = eclipse_state_->getIOConfig();
+                ioConfig->setOutputDir(output_dir_);
```

Comment:
> I don't understand it ...

Sometimes seemingly `const` pointers are passed, but the object pointed to is mutable. I've removed some those seemingly-const-but-not really interactions in opm-parser in the past.

>      

I would actually vote for mutable here; as a general rule I think the EclipseState object should be const for the entire lifetime - however there are some aspects which we want to override based on commandline switches which take presedence over the deck, as in this case, since these are non-standard I think it is OK to use mutable

Personally I disagree here. It's a visible change to a `const` object which is a huge no-no. If you want that behaviour, make it non-const. A solutionis to make it cheap to move+modify and create a **new** EclipseState from the old one, with the configuration changes applied.